### PR TITLE
Update upgrading search #6385

### PR DESCRIPTION
**Pull request by alhambrav**
### Ticket reference or full description of what's in the PR
Update upgrading search section for:
- upgrading 3.1.x to 4.2.0 
- upgrading 4.0.x to 4.2.0 
- remove the section upgrading 4.1.x to 4.2.0 since the [error](https://github.com/opensearch-project/observability/issues/1771) that appears is now resolved in the OpenSearch version used in 4.2.0 

https://github.com/craftercms/craftercms/issues/6385

### CODE REVIEW

**Review by sumerjabri — CHANGES_REQUESTED** [review]
What about upgrading from 4.1 to 4.2? Even if there is nothing to do, we should have a section that says that.

**Review by sumerjabri — APPROVED** [review]

**Comment by sumerjabri** [comment]
@yacdaniel bump

File: source/by-role/system-admin/upgrade/upgrading-search.rst [hunk 1]
```diff
@@ -54,35 +54,103 @@ Here is the ``upgrade-search`` script params:
...
+       ...
+       > Backup the bin folder before upgrade? [(Y)es/(N)o]:n
+       ...
+       ========================================================================
+       Upgrade completed
+       ========================================================================
+       !!! Please read the release notes and make any necessary manual changes, then run the post upgrade script: /path/of/install/to/be/upgraded/bin/upgrade/post-upgrade.sh !!!
 
-#. Run the ``post-upgrade.sh`` script from the install that's being upgraded.  Before starting CrafterCMS, you'll need to configure the installation root directory to use Java version 17.  Remember to read the release notes or any relevant upgrade articles and make any necessary manual changes before running the `post-upgrade.sh`` script
+#. Before performing the post-upgrade, please upgrade the DB in the install that's being upgraded.
```
Comment (yacdaniel) [inline source/by-role/system-admin/upgrade/upgrading-search.rst:95]: @jmendeza isn't the `upgradedb` now removed from the upgrade scripts and done automatically? related to https://github.com/craftersoftware/craftercms/issues/1010

Comment (jmendeza) [reply source/by-role/system-admin/upgrade/upgrading-search.rst:95]: Yes @yacdaniel, DB upgrade would happen on startup.

**Review by yacdaniel — APPROVED** [review]

**Pull request merged**